src/classes/HH_Container_LCTRL.cls

```diff
@@ -446,23 +425,62 @@ public with sharing class HH_Container_LCTRL {
     */
     @AuraEnabled
     public static void saveHouseholdPage(SObject hh, list<Contact> listCon, list<Contact> listConRemove, list<Account>listHHMerge) {
-
-        // save the Household
         updateHousehold(hh);
         
         // need to merge any households (Accounts only) before we save contacts
         // so we avoid deleting a household if that contact was the last one in the hh.
-        if (listHHMerge != null && listHHMerge.size() > 0) {
+        if (isNotEmpty(listHHMerge)) {
             mergeHouseholds((Account)hh, listHHMerge);
         }
+
+        List<Contact> contacts = new List<Contact>(listCon);
+        contacts.addAll(listConRemove);        
+        upsertContacts(contacts);  
         
-        // save the contacts
-        upsertContacts(listCon);
+        if (isNotEmpty(listHHMerge)) {
+            cleanupAddresses(new List<Id>{ (Id) hh.get('Id') });
+        }          
+    }    
+
+    /*******************************************************************************************************
+    * @description Checks if the Account list is not empty
+    * @param accounts The list of Accounts
+    * @return Boolean True is the list is not empty, false if the list is null or empty
+    */
+    static Boolean isNotEmpty(List<Account> accounts) {
+        return accounts != null && !accounts.isEmpty();
+    }
+
+    /*******************************************************************************************************
+    * @description Cleans up/refreshes the Addresses for specified Account Ids
+    * @param accountIds List of Account Ids
+    * @return void
+    */
+    static void cleanupAddresses(List<Id> accountIds) {
```

Comment:
The code was written so it is easy to extract and refactor if/when we decide to use a common method to process Address cleanup (see ACCT_AccountMerge_TDTM.handleAccountMergeFixups() for another call to cleanup Addresses). However, I can easily make the change to the method to accept one Account Id instead of the bulk. Please let me know.
Methods in Apex are private by default, so no need to explicitly specify it.